src/V3Sched.cpp

```diff
@@ -842,16 +857,42 @@ void createEval(AstNetlist* netlistp,  //
         textp->text(VString::replaceWord(textp->text(), "act", "nba"));
     });
 
+    const auto makeTrigClear = [=](AstVarScope* trigVscp) {
+        AstVarRef* const refp = new AstVarRef{flp, trigVscp, VAccess::WRITE};
+        AstCMethodHard* const callp = new AstCMethodHard{flp, refp, "clear"};
+        callp->statement(true);
+        callp->dtypeSetVoid();
+        return callp;
+    };
+
     // Create the active eval loop
     AstNode* const activeEvalLoopp
         = makeEvalLoop(
               netlistp, "act", "Active", actTrig.m_vscp, actTrig.m_dumpp,
               [&]() {  // Trigger
-                  return new AstCCall{flp, actTrig.m_funcp};
+                  AstNode* resultp = nullptr;
+                  if (timingKit.m_delayResumep) {
+                      // Resume delay scheduler
+                      resultp = timingKit.m_delayResumep;
+                  }
+
+                  resultp = AstNode::addNext(resultp, new AstCCall{flp, actTrig.m_funcp});
+
+                  if (timingKit.m_trigVscp) {
+                      // Reset timing triggers
+                      resultp = AstNode::addNext(resultp, makeTrigClear(timingKit.m_trigVscp));
+                  }
```

Comment:
Isn't this redundant given we add this into the actTrig function as well?

Reply:
Oops, I think I removed it and then undid it.